src/Schema/Column.php

```diff
@@ -190,7 +190,7 @@ public function setNotnull($notnull)
     }
 
     /**
-     * @param mixed $default
+     * @param string|null $default
```

Comment:
The default property only accept `string|null`
```
    /** @var string|null */
    protected $_default;
```

Reply:
Technically you're right and I agree with this restriction but it will result in a BC break in 4.0 (which is also fine). For instance: https://github.com/doctrine/dbal/blob/bd8547f6a4a20c1a5780dd031bbf8cddbc218b59/tests/Functional/Schema/SQLServerSchemaManagerTest.php#L83

I'd like all these issues addressed in `3.4.x` before the merge up.

This would also require changes in documentation. For instance: https://github.com/doctrine/dbal/blob/f57646665700d833054d1ac597cd28a19e9f35a3/docs/en/reference/schema-representation.rst#L91-L92

And the nastiest part is that's not no longer clear what exactly the meaning of the column default is: previously it was mixed, so it implied that it's the default _value_ (e.g. `(int) 666`). Now we are saying it can only be a string, so we imply that it's SQL but we don't support arbitrary SQL expressions.

Reply:
Should I split the PR in two then to not delay the `setColumnDefinition` phpdoc update. (which is the only one impacting me currently) ?

Reply:
That should work.

Reply:
Ok, I removed this changed and renamed the commit/PR.

Does an issue track the change needed for the default property ?

Reply:
I believe it’s already tracked somewhere. Unless you plan on working on it, no need to file a new one.